activation/nipost.go

```diff
@@ -530,6 +573,10 @@ func (nb *NIPostBuilder) getBestProof(
 
 	var eg errgroup.Group
 	for _, r := range registrations {
+		if len(r.RoundID) == 0 {
+			continue // skip failed registrations
+		}
```

Comment:
It's incorrect as the first poet round has ID 0. IMHO, it's better not to give existing fields "extra meaning" but to have a dedicated field.